Configurations.md

```diff
@@ -0,0 +1,1760 @@
+    where T: Eq
+{
+    sit: T,
+}
+```
+
+#### `"PreferSameLine"`:
+
+```rust
+struct Lorem {
+    ipsum: bool,
+}
+
+struct Dolor<T>
+    where T: Eq {
+    sit: T,
+}
+```
+
+#### `"SameLineWhere"`:
+
+```rust
+struct Lorem {
+    ipsum: bool,
+}
+
+struct Dolor<T>
+    where T: Eq
+{
+    sit: T,
+}
+```
+
+## `match_block_trailing_comma`
+
+Put a trailing comma after a block based match arm (non-block arms are not affected)
+
+- **Default value**: `false`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+match lorem {
+    Lorem::Ipsum => {
+        println!("ipsum");
+    }
+    Lorem::Dolor => println!("dolor"),
+}
+```
+
+#### `true`:
+
+```rust
+match lorem {
+    Lorem::Ipsum => {
+        println!("ipsum");
+    },
+    Lorem::Dolor => println!("dolor"),
+}
+```
+
+See also: [`indent_match_arms`](#indent_match_arms), [`trailing_comma`](#trailing_comma), [`wrap_match_arms`](#wrap_match_arms).
+
+## `max_width`
+
+Maximum width of each line
+
+- **Default value**: `100`
+- **Possible values**: any positive integer
+
+See also [`error_on_line_overflow`](#error_on_line_overflow).
+
+## `newline_style`
+
+Unix or Windows line endings
+
+- **Default value**: `"Unix"`
+- **Possible values**: `"Native"`, `"Unix"`, `"Windows"`
+
+## `normalize_comments`
+
+Convert /* */ comments to // comments where possible
+
+- **Default value**: `false`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+// Lorem ipsum:
+fn dolor() -> usize {}
+
+/* sit amet: */
+fn adipiscing() -> usize {}
+```
+
+#### `true`:
+
+```rust
+// Lorem ipsum:
+fn dolor() -> usize {}
+
+// sit amet:
+fn adipiscing() -> usize {}
+```
+
+## `reorder_imported_names`
+
+Reorder lists of names in import statements alphabetically
+
+- **Default value**: `false`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+use super::{lorem, ipsum, dolor, sit};
+```
+
+#### `true`:
+
+```rust
+use super::{dolor, ipsum, lorem, sit};
+```
+
+See also [`reorder_imports`](#reorder_imports).
+
+## `reorder_imports`
+
+Reorder import statements alphabetically
+
+- **Default value**: `false`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+use lorem;
+use ipsum;
+use dolor;
+use sit;
+```
+
+#### `true`:
+
+```rust
+use dolor;
+use ipsum;
+use lorem;
+use sit;
+```
+
+See also [`reorder_imported_names`](#reorder_imported_names).
+
+## `single_line_if_else_max_width`
+
+Maximum line length for single line if-else expressions. A value of zero means always break if-else expressions.
+
+- **Default value**: `50`
+- **Possible values**: any positive integer
+
+#### Lines shorter than `single_line_if_else_max_width`:
+```rust
+let lorem = if ipsum { dolor } else { sit };
+```
+
+#### Lines longer than `single_line_if_else_max_width`:
+```rust
+let lorem = if ipsum {
+    dolor
+} else {
+    sit
+};
+```
+
+See also: [`control_brace_style`](#control_brace_style).
+
+## `skip_children`
+
+Don't reformat out of line modules
```

Comment:
inline = `mod { ... }`, out of line = `mod foo;` where foo.rs (or whatever) has the module contents